### PR TITLE
Fix signer id use in stackerdb

### PR DESCRIPTION
**Pull request by jferrant**
In my massive PR, I aaccidentally conflated signer id and signer slot id. this did not break tests because we happened to have everything line up (odds are we ALWAYS will have everything line up since the reward set is used to buidl stackerdb, but this will make sure we do not rely on this)

### CODE REVIEW

File: stacks-signer/src/client/mod.rs [hunk 1]
```diff
@@ -459,7 +456,7 @@ pub(crate) mod tests {
                     &StacksPublicKey::from_slice(ecdsa_public_key.to_bytes().as_slice())
                         .expect("Failed to create stacks public key"),
                 );
-                signer_slot_ids.insert(address, signer_id); // Note in a real world situation, these would not always match
+                signer_slot_ids.push(signer_id); // Note in a real world situation, these would not always match
```
Comment (xoloki) [inline stacks-signer/src/client/mod.rs:459]: Can you update the comment?  Previously it looked like the comment meant that `address` and `signer_id` would not always match.

File: stacks-signer/src/client/mod.rs [hunk 2]
```diff
@@ -486,7 +483,7 @@ pub(crate) mod tests {
                 &StacksPublicKey::from_slice(public_key.to_bytes().as_slice())
                     .expect("Failed to create stacks public key"),
             );
-            signer_slot_ids.insert(address, signer_id); // Note in a real world situation, these would not always match
+            signer_slot_ids.push(signer_id); // Note in a real world situation, these would not always match
```
Comment (xoloki) [inline stacks-signer/src/client/mod.rs:486]: Same

**Review by xoloki — DISMISSED** [review]
LGTM, one small question about a comment.

File: stacks-signer/src/client/mod.rs (hunk 1 above)
Comment (jferrant) [reply stacks-signer/src/client/mod.rs:459]: No this comment is accurate. It means that signer id and signer slot id would not always match

File: stacks-signer/src/client/mod.rs (hunk 2 above)
Comment (jferrant) [reply stacks-signer/src/client/mod.rs:486]: See other comment.

**Review by hstove — APPROVED** [review]

**Review by kantai — DISMISSED** [review]

**Review by kantai — APPROVED** [review]

**Pull request merged**